cmd/hive/clustersync_test.go

```diff
@@ -0,0 +1,396 @@
+package hive
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/mock/gomock"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+	"github.com/openshift/hive/apis/hiveinternal/v1alpha1"
+	mockk8s "github.com/openshift/osdctl/cmd/hive/clusterdeployment/mock/k8s"
+)
+
+// Mock data for ClusterDeployment and ClusterSync that are used across all test cases
+var (
+	cdList = hivev1.ClusterDeploymentList{
```

Comment:
It looks like these global vars are only being used by the `setupMockClient` function.  Can you either inline them or put them in a private function that the `setupMockClient` would call and have the lists returned to it?